src/pages/Verification/Confirm.tsx

```diff
@@ -0,0 +1,96 @@
+import { useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { Typography, CircularProgress } from '@mui/material';
+import { makeStyles } from '@mui/styles';
+import axios, { AxiosResponse } from 'axios';
+import { BASE_URL } from '@/config/default';
+import { selectToken } from "@/redux/userSlice";
+import { useSelector } from "react-redux";
+import { RootState } from '@/redux/store';
+
+
+interface AuthResponse {
+  data?: any;
+  error?: string;
+  message?: string;
+}
+
+const useStyles = makeStyles(() => ({
+  root: {
+    display: 'flex',
+    flexDirection: 'column',
+    alignItems: 'center',
+    justifyContent: 'center',
+    minHeight: '100vh',
+  },
+  message: {
+    marginTop: "2px",
+  },
+}));
+
+function Confirm() {
+  const classes = useStyles();
+  const navigate = useNavigate();
+  const [confirmationStatus, setConfirmationStatus] = useState('pending');
+  const authtoken = useSelector((state: RootState) => selectToken(state));
+  //useParam
+  const token = window.location.pathname.split('/').pop();
+  
+  useEffect(() => {
+    const verifyToken = async () => {
+      try {
+        const response: AxiosResponse<AuthResponse> = await axios.get(`${BASE_URL}/api/auth/confirm/${token}`, 
+        { withCredentials: true, headers: { Authorization: `Bearer ${authtoken}` }}
+        );
+        
+        if (response.status === 200) {
+          setConfirmationStatus('success');
+        } else {
+          setConfirmationStatus('error');
+        }
+      } catch (error) {
+        setConfirmationStatus('error');
+      }
+    };
+
+    verifyToken();
+  }, [token]);
+
+  useEffect(() => {
+    if (confirmationStatus === 'success') {
+      navigate('/DAOMatcher');
+    }
+  }, [confirmationStatus, navigate]);
+
+  let confirmationMessage;
```

Comment:
Would be nice to rename this variable in a way that it can indicate it is a component. It might be confused with a string when reading the code?

Reply:
how about "ConfirmationMessageComponent"